src/resources/packages/wizard/components/tabs.js

```diff
@@ -10,112 +10,216 @@ import * as TicketsContent from './tabs/tickets';
+	 * @property {string}   baseClass - The base class name for the tab. This remains unchanged for reference.
+	 * @property {boolean}  disabled  - Whether the tab is disabled or not.
+	 * @property {function} content   - The content to render for the tab.
+	 */
 	const initialTabs = [
 		{
 			name: 'intro',
 			title: __('Intro', 'the-events-calendar'),
 			className: 'tec-events-onboarding__tab-intro',
+			baseClass: 'tec-events-onboarding__tab-intro',
 			disabled: false,
-			content: IntroContent,
-			current: true
+			content: IntroContent
 		},
 		{
 			name: 'display',
 			title: __('Calendar Display', 'the-events-calendar'),
 			className: 'tec-events-onboarding__tab-display',
+			baseClass: 'tec-events-onboarding__tab-display',
 			disabled: true,
-			content: DisplayContent,
-			current: false
+			content: DisplayContent
 		},
 		{
 			name: 'settings',
 			title: __('Event Settings', 'the-events-calendar'),
 			className: 'tec-events-onboarding__tab-settings',
+			baseClass: 'tec-events-onboarding__tab-settings',
 			disabled: true,
-			content: SettingsContent,
-			current: false
+			content: SettingsContent
 		},
 		{
 			name: 'organizer',
 			title: __('Event Organizer', 'the-events-calendar'),
 			className: 'tec-events-onboarding__tab-organizer',
+			baseClass: 'tec-events-onboarding__tab-organizer',
 			disabled: true,
-			content: OrganizerContent,
-			current: false
+			content: OrganizerContent
 		},
 		{
 			name: 'venue',
 			title: __('Event Venue', 'the-events-calendar'),
 			className: 'tec-events-onboarding__tab-venue',
+			baseClass: 'tec-events-onboarding__tab-venue',
 			disabled: true,
-			content: VenueContent,
-			current: false
+			content: VenueContent
 		},
 		{
 			name: 'tickets',
 			title: __('Tickets', 'the-events-calendar'),
 			className: 'tec-events-onboarding__tab-tickets',
+			baseClass: 'tec-events-onboarding__tab-tickets',
 			disabled: true,
-			content: TicketsContent,
-			current: false
+			content: TicketsContent
 		},
 	];
 
+	/**
+	 * State management for tabs.
+	 *
+	 * @property {Array} tabs - The current state of tabs.
+	 */
 	const [tabs, setTabs] = useState(initialTabs);
+
+	/**
+	 * State management for active tab.
+	 *
+	 * @property {string} activeTab - The currently active tab.
+	 * @default 'intro'
+	 */
 	const [activeTab, setActiveTab] = useState('intro');
 
+	/**
+	 * Class names for completed tab state.
+	 *
+	 * @type {string}
+	 */
+	const completedClass = "tec-events-onboarding__tab--completed";
+
+	/**
+	 * Class names for active tab state.
+	 *
+	 * @type {string}
+	 */
+	const activeClass = "tec-events-onboarding__tab--active";
+
+	/**
+	 * "Clean" the tab classes by removing the "active" class from all tabs.
+	 */
+	const cleanTabClasses = () => {
+		const cleanedTabs = tabs.map(tab => (
+			// We just want to remove the "active" class from all tabs prior to setting the next one.
+			tab.className = tab.className.replace(activeClass, "")
+		));
```

Comment:
```suggestion
		const cleanedTabs = tabs.map(tab => ( {
			...tag,
			// We just want to remove the "active" class from all tabs prior to setting the next one.
			className: tab.className.replace(activeClass, "")
		));
```

You're modifying the `tab` instance, which you don't want to do. React relies on the object identity to detect changes. Instead, you'll want to `map` and return a new object.